OpenSim/Common/Component.cpp

```diff
@@ -1098,19 +1099,27 @@ void Component::markPropertiesAsSubcomponents()
 
 // mark a Component as a subcomponent of this one. If already a
 // subcomponent, it is not added to the list again.
-void Component::markAsPropertySubcomponent(Component* component)
+void Component::markAsPropertySubcomponent(const Component* component)
 {
     // Only add if the component is not already a part of this Component
     // So, add if empty
-    SimTK::ReferencePtr<Component> compRef(component);
+    SimTK::ReferencePtr<Component> compRef(const_cast<Component*>(component));
     if (_propertySubcomponents.empty() ){
-        _propertySubcomponents.push_back(SimTK::ReferencePtr<Component>(component));
+        // Must reconstruct the reference pointer in place in order
```

Comment:
Please excuse my ignorance, but why is this outer `if` necessary? Does `std::find(...)` (line 1116) throw an exception if `_propertySubcomponents` is empty?